clients/src/main/java/org/apache/kafka/clients/admin/DescribeTopicsOptions.java

```diff
@@ -47,8 +49,32 @@ public DescribeTopicsOptions includeAuthorizedOperations(boolean includeAuthoriz
         return this;
     }
 
+    /**
+     * Whether to use the DescribeTopicPartitions API. It should be set to false if DescribeTopicPartitions API is
+     * not supported.
+     *
+     */
+    public DescribeTopicsOptions useDescribeTopicPartitionsApi(boolean useDescribeTopicPartitionsApi) {
+        this.useDescribeTopicPartitionsApi = useDescribeTopicPartitionsApi;
+        return this;
+    }
+
+    // Note that, partitionSizeLimitPerResponse will not be effective if it is larger than the config
+    // max.request.partition.size.limit on the server side.
+    public DescribeTopicsOptions partitionSizeLimitPerResponse(int partitionSizeLimitPerResponse) {
+        this.partitionSizeLimitPerResponse = partitionSizeLimitPerResponse;
+        return this;
+    }
+
     public boolean includeAuthorizedOperations() {
         return includeAuthorizedOperations;
     }
 
+    public boolean useDescribeTopicPartitionsApi() {
```

Comment:
I suggest just `useDescribeTopicPartitions()`. In the Javadoc, you can mention that it's using the DescribeTopicPartitions API under the covers. Most users of the Kafka admin client would consider `KafkaAdminClient` to be the API, rather than the Kafka protocol which is what is meant here.

Reply:
removed.